### ISSUE TITLE
Missing index entries

### ISSUE THREAD

**Issue by drj826** [drj826/SML#2]
There are index entries missing in the rendered default HTML index of the SML User Guide.  For instance the entry for "table!cell!color" is missing. There are two index entries near one another in the manuscript: "table!cell!color" and then "table!cell!justification." The second one is in the index but the first is missing.

**Comment by drj826** [comment]
There are two index entries near one another in the manuscript: "table!cell!color" and then "table!cell!justification."  The second one is in the index but the first is missing.

**Comment by drj826** [comment]
While working on this issue today I uncovered a related, more serious, issue.  My parser implementation was adding index entries to the index as soon as it detected the end of the index entry.  The problem with this design is that SML::Entity divisions are often parsed BEFORE the document that contains them.  The parser tried to add the index entries to a non-existent document index and they were effectively ignored because I didn't raise an error on this condition.

This problem is now fixed.  I refactored the code that builds document indexes to make it more understandable.

The original problem, missing index entries, still remains.
